go/vt/topo/zk2topo/server.go

```diff
@@ -100,14 +107,32 @@ func (zs *Server) connForCell(ctx context.Context, cell string) (Conn, string, e
 	if err := proto.Unmarshal(data, ci); err != nil {
 		return nil, "", fmt.Errorf("cannot Unmarshal CellInfo for cell %v: %v", cell, err)
 	}
+	serverAddress := cellOverride(cell)
+	if serverAddress == "" {
+		serverAddress = ci.ServerAddress
+	}
 	ins = &instance{
 		root: ci.Root,
-		conn: Connect(ci.ServerAddress),
+		conn: Connect(serverAddress),
 	}
 	zs.instances[cell] = ins
 	return ins.conn, ins.root, nil
 }
 
+func cellOverride(cell string) string {
```

Comment:
Might be a better way to do this? But that can be done post checkin.

Reply:
Jon is right that Go is not super elegant in this regard. Something like this is about as good as you can hope for:

```go
func cellOverride2(cell string) string {
	i := strings.Index(cellOverrides, cell + "=")
	if i == -1 {
		return ""
	}
	i += len(cell)+1
	j := strings.Index(cellOverrides[i:], ",")
	if j == -1 {
		j = len(cellOverrides) - i
	}
	return cellOverrides[i:i+j]
}
```